File: src/main/java/com/tom/student/Student.java
package com.tom.student;

import java.sql.SQLOutput;

public class Student {
    String id;
    String name;
    int english;
    int math;
    public Student(String name, int english, int math)
    {
        this.name = name;
        this.english = english;
        this.math=math;
    }

    public int highest(){
        int max = (english>math)?english:math;
       /* if(english>math){
            max = english;
        }else{
            max = math;
        }*/
        return max;
    }

    public void print(){
//        sout+tab
        System.out.println(name+"\t" +english+"\t"+math+
                "\t"+(english+math)/2);
    }
}

File: src/main/kotlin/com/tom/student/Student.kt
package com.tom.kotlin

import java.util.*

fun main()
{
//    userInput()
    val stu = Student("Hank",80,90)
    stu.print()
    var test =123
    println("Test is : $test")
    println("High score:${stu.highest()}")
}

class Student(var name:String?,var english:Int,var math:Int){
    fun print(){
        println(name+"\t"+english+"\t"+math+"\t"+(english+math)/2)
    }

    fun highest() : Int{
        var max = if (english>math){
            println("english")
            english
        }else {
            println("math")
            math
        }
        return max
    }

    fun nameCheck() {
        println(name?.length)
    }
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
//    print("Please enter student's name:")
//    var name = scanner.next()
    var name = null
    print("Please enter student's english:")
    var english = scanner.nextInt()
    print("Please enter student's math:")
    var math = scanner.nextInt()
    val stu = Student(name, english, math)
    stu.print()
    stu.nameCheck()
}
